Shallow water: Add support for geotiff image files for reading in a bathymetry

tamiko opened the pull request:
This is a quick hack to add rudimentary support for geotiff files. We could add support for geotiff (and other raster elevation formats supported by gdal) much more comprehensively, I'll open an issue for that.

tamiko committed:
CMake: add support for gdal library

tamiko committed:
ShallowWater: add geotiff skeleton

tamiko committed:
ShallowWater: GeoTIFF: add transformation, finish up bathymetry

tamiko committed:
Fix compilation with deal.II version 9.4.2

ejtovar commented on source/shallow_water/initial_state_geotiff.h:
```diff
@@ -0,0 +1,395 @@
+        this->add_parameter(
+            "filename", filename_, "GeoTIFF: image file to load");
+
+        transformation_ = {0., 0.01, 0., 0., 0., 0.01};
```
Out of curiosity, where did these numbers come from?

tamiko replied:
The numbers are simply meant to have a nonsingular transformation - no deeper meaning. I chose to scale the x and y axis by 0.01 so that the total dimension of the domain does not become too large (which might be unexpected).

ejtovar reviewed (approved):
This looks good! Should we add a url/reference in the `initial_state_geotiff.h` file to the  GDAL documentation?

tamiko committed:
ShallowWater: update documentation

tamiko commented:
@ejtovar Good point. I have updated the class description and added a link. Eventually we will need to provide much more documentation on how to use geotiff's (and on how to use ryujin in the first place). Let's address this in a follow-up PR.
